packages/contracts-bedrock/contracts/libraries/Nibbles.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+/**
+ * @title Nibbles
+ * @notice Nibbles is a simple library for dealing with nibble arrays.
+ */
+library Nibbles {
+    /**
+     * @notice Converts a byte array into a nibble array by splitting each byte into two nibbles.
+     *         Resulting nibble array will be exactly twice as long as the input byte array.
+     *
+     * @param _bytes Input byte array to convert.
+     *
+     * @return Resulting nibble array.
+     */
+    function toNibbles(bytes memory _bytes) internal pure returns (bytes memory) {
+        bytes memory nibbles = new bytes(_bytes.length * 2);
```

Comment:
I'd like to follow up with a PR that caches the length and increments in an unchecked block